Assets/Scripts/UI/Canvas/StateMachine/UIStates.cs

```diff
@@ -0,0 +1,17 @@
+using UnityEngine;
+
+namespace UI.Canvas.StateMachine
+{
+    public class StateChange : MonoBehaviour
+    {
+        public UIStates states;
```

Comment:
Het ding dat zoiezo anders moet is dat de `s` weg moet, het is 1 state en niet meerdere.
```suggestion
        public UIState state;
```

Dit is alleen public van wege [deze line 15](Assets/Scripts/UI/Canvas/StateMachine/UIStateCaller.cs). Hier kan makkelijk een getter voor gemaakt worden.

```suggestion
        [SerializeField] private UIState state;
        
        public UIState State => state;
```
OF
```suggestion
        [field: SerializeField] public UIState State{ get; private set; }
```